src/topology/algebra/ordered/basic.lean

```diff
@@ -1494,14 +1494,17 @@ local notation `|` x `|` := abs x
 lemma nhds_eq_infi_abs_sub (a : α) : 𝓝 a = (⨅r>0, 𝓟 {b | |a - b| < r}) :=
 begin
   simp only [le_antisymm_iff, nhds_eq_order, le_inf_iff, le_infi_iff, le_principal_iff, mem_Ioi,
-    mem_Iio, abs_sub_lt_iff, @sub_lt_iff_lt_add _ _ _ _ a, @sub_lt _ _ a, set_of_and],
+    mem_Iio, abs_sub_lt_iff, @sub_lt_iff_lt_add' _ _ _ _ a, @sub_lt _ _ _ _ a, set_of_and],
   refine ⟨_, _, _⟩,
   { intros ε ε0,
-    exact inter_mem_inf_sets
-      (mem_infi_sets (a - ε) $ mem_infi_sets (sub_lt_self a ε0) (mem_principal_self _))
-      (mem_infi_sets (ε + a) $ mem_infi_sets (by simpa) (mem_principal_self _)) },
+    refine inter_mem_inf_sets
+      (mem_infi_sets (a - ε) $ mem_infi_sets (sub_lt_self a ε0) _)
+      (mem_infi_sets (ε + a) $ mem_infi_sets (by simpa) _),
+    refine mem_principal_sets.mp (λ h j, sub_lt_iff_lt_add'.mp (mem_Ioi.mp j)),
+    exact mem_principal_sets.mp (λ x, sub_lt_iff_lt_add.mpr) },
```

Comment:
Why does `mem_principal_self` no longer work here?

Reply:
I do not know: I tried to keep all the `simp` lemmas, but they do not seem to fire here.  Could it be because now they have to go through an implicit instance?

Reply:
You forgot to add two underscores to `sub_lt_iff_lt_add`. 
Replace the proof by
```
lemma nhds_eq_infi_abs_sub (a : α) : 𝓝 a = (⨅r>0, 𝓟 {b | |a - b| < r}) :=
begin
  simp only [le_antisymm_iff, nhds_eq_order, le_inf_iff, le_infi_iff, le_principal_iff, mem_Ioi,
    mem_Iio, abs_sub_lt_iff, @sub_lt_iff_lt_add _ _ _ _ _ _ a, @sub_lt _ _ _ _ a, set_of_and],
  refine ⟨_, _, _⟩,
  { intros ε ε0,
    refine inter_mem_inf_sets
      (mem_infi_sets (a - ε) $ mem_infi_sets (sub_lt_self a ε0) (mem_principal_self _))
      (mem_infi_sets (ε + a) $ mem_infi_sets (by simpa) (mem_principal_self _)) },
  { intros b hb,
    exact mem_infi_sets (a - b) (mem_infi_sets (sub_pos.2 hb) (by simp [Ioi])) },
  { intros b hb,
    exact mem_infi_sets (b - a) (mem_infi_sets (sub_pos.2 hb) (by simp [Iio])) }
end
```

Reply:
Floris, thank you very much for figuring this out: I was convinced that it was an issue with `simp` lemmas and could not understand what the issue was!